### ISSUE TITLE
Error when reordering plugin sorter

### ISSUE THREAD

**Issue by jni** [napari/napari#3131]
## 🐛 Bug

When I tried to drag and drop plugins for get_reader in the plugin sorter, I got the following error:

```pytb
AttributeError                            Traceback (most recent call last)
~/projects/napari/napari/_qt/widgets/qt_plugin_sorter.py in dropEvent(self=<napari._qt.widgets.qt_plugin_sorter.QtHookImplementationListWidget object>, event=<PyQt5.QtGui.QDropEvent object>)
    227         """
    228         super().dropEvent(event)
--> 229         order = [self.item(r).hook_implementation for r in range(self.count())]
        order = undefined
        self.item = <built-in method item of QtHookImplementationListWidget object at 0x17c5d7d30>
        global r.hook_implementation = undefined
        global r = undefined
        global range = undefined
        self.count = <built-in method count of QtHookImplementationListWidget object at 0x17c5d7d30>
    230         self.order_changed.emit(order)
    231

~/projects/napari/napari/_qt/widgets/qt_plugin_sorter.py in <listcomp>(.0=<range_iterator object>)
    227         """
    228         super().dropEvent(event)
--> 229         order = [self.item(r).hook_implementation for r in range(self.count())]
        global order = undefined
        global self.item = undefined
        r.hook_implementation = undefined
        r = 4
        global range = undefined
        global self.count = undefined
    230         self.order_changed.emit(order)
    231

AttributeError: 'QListWidgetItem' object has no attribute 'hook_implementation'
```

**Comment by tlambert03** [comment]
thanks for the report @jni.  I'm unable to recreate at the moment... but I can see how this might potentially happen if something independently added an item to `QtHookImplementationListWidget`.  We might be able to patch it just by using `getattr(..., None)`... but i'd feel a bit better about it if i knew exactly how a `QListWidgetItem` got added without getting that `hook_implementation` attribute (which should have happened [here](https://github.com/napari/napari/blob/702bb6e0a9db20c278547fc086ed49b663354a36/napari/_qt/widgets/qt_plugin_sorter.py#L212))

**Comment by jni** [comment]
... aaaand I can't reproduce either. 🤦 

A bit more context, though I don't know how much it would help:

- I installed napari-bioformats
- I tried to open a .nd2 file
- The builtins keep higher priority because the author of napari-bioformats has no faith in their own library. 😜 
- The builtins don't error out gracefully, so I was getting annoying error messages (though eventually bioformats picks it up)
- I tried to reorder the plugins.

Unfortunately I can't remember whether I tried to drag plugins down or bioformats up. ALSO, and this probably *is* relevant, when I fiddled around some more there turned out to be a blank item in the list — ie dragging *way* down left a gap in the list between the last item and the newly dragged item.

Perhaps recapitulating all of the above in a clean env would work... 🤷

**Comment by tlambert03** [comment]
> there turned out to be a blank item in the list

Ohh!  Yeah that's helpful.  Thanks for the details, will dig.

Also, we should definitely fix the annoying error message independently

**Comment by ppwadhwa** [comment]
Is this issue still a problem?   I just created a fresh environment, installed napari from github, conda installed napari-bioformats, and found a sample .nd2 file to open.  It worked for me!  I saw that the bioformats reader was also listed at the top when I looked at the plugins list.  I moved things around just for fun and still no problem.  Let me know if I should keep digging here.  I also didn't see any error messages.
